RMS/Astrometry/FFTalign.py

```diff
@@ -141,38 +232,101 @@ def findStarsTransform(config, reference_list, moved_list, img_size=256, dot_rad
     moved_list[:, 1] += shift_y
 
     # Construct the reference and moved images
-    img_ref = constructImage(img_size, reference_list, dot_radius)
-    img_mov = constructImage(img_size, moved_list, dot_radius)
-
+    img_ref = constructImage(img_size, reference_list, dot_radius).astype(np.float64)
+    img_mov = constructImage(img_size, moved_list, dot_radius).astype(np.float64)
 
-    # Run the FFT registration
+    # Run the FFT registration using custom imreg_dft-style algorithm
     try:
-        res = imreg_dft.imreg.similarity(img_ref, img_mov)
-
-    except ValueError:
-        log.warning('imreg_dft error: The scale correction is too high!')
+        # Apply Hann window to reduce spectral leakage from image borders
+        winhann = window('hann', img_ref.shape)
+        img_ref_windowed = img_ref * winhann
+        img_mov_windowed = img_mov * winhann
+
+        # Compute FFT magnitude spectra (translation-invariant representation)
+        dft_ref = fftshift(fft2(img_ref_windowed))
+        dft_mov = fftshift(fft2(img_mov_windowed))
+
+        # Apply log-polar filter to suppress low frequencies (essential for rotation accuracy)
+        filt = logpolarFilter(img_ref.shape)
+        dft_ref_filt = dft_ref * filt
+        dft_mov_filt = dft_mov * filt
+
+        # Apply custom log-polar transform
+        pcorr_shape = (int(max(img_ref.shape)),) * 2
+        log_base = getLogBase(img_ref.shape, pcorr_shape[1])
+
+        lp_ref = logpolarTransform(np.abs(dft_ref_filt), pcorr_shape, log_base)
+        lp_mov = logpolarTransform(np.abs(dft_mov_filt), pcorr_shape, log_base)
+
+        # Find rotation and scale via phase correlation on log-polar images
+        shifts = phaseCorrelationSubpixel(lp_ref, lp_mov)
+        arg_ang, arg_rad = shifts
+
+        # Convert shifts to rotation angle and scale factor
+        angle = -np.pi * arg_ang / float(pcorr_shape[0])
+        angle = np.rad2deg(angle)
+        angle = ((angle + 180) % 360) - 180  # Normalize to [-180, 180]
+
+        scale = log_base ** arg_rad
+
+        # Invert to get the transform from reference to moved (matching imreg_dft convention)
+        angle = -angle
+        scale = 1.0 / scale
+
+        # Check for unreasonable scale values
+        if scale < 0.5 or scale > 2.0:
+            log.warning('FFT registration error: The scale correction is too high ({:.3f})!'.format(scale))
+            return 0.0, 1.0, 0.0, 0.0
+
+        # Apply rotation correction to moved image to find translation
+        # Use skimage.transform.rotate which handles center rotation properly
+        img_mov_corrected = rotate(img_mov, angle, resize=False, mode='constant', cval=0, order=1)
+
+        # Handle scale correction if significant
+        if abs(scale - 1.0) > 0.001:
+            from scipy.ndimage import zoom
```

Comment:
Recommend moving the import to the top of the file with other imports, to follow PEP-08 best practice and prevent confusion / multiple inclusion of the same module.